### PR TITLE
[IMP] web, *: unify sprintf & gettext

### PR DESCRIPTION
**Pull request by oomsveta**
Improve `gettext` to directly handle value injection within translations, removing the need for `sprintf`.

```js
sprintf(_t("Due in %s days"), days);
```
becomes
```js
_t("Due in %s days", days);
```

Enterprise: https://github.com/odoo/enterprise/pull/45370

### CODE REVIEW

File: addons/web/static/src/core/l10n/translation.js [hunk 1]
```diff
@@ -17,9 +19,13 @@ export const translationIsReady = new Deferred();
...
+export function _t(term, ...values) {
     if (translatedTerms[translationLoaded]) {
-        return translatedTerms[term] || term;
+        const translation = translatedTerms[term] ?? term;
+        if (values.length === 0) {
+            return translation;
+        }
+        return sprintf(translation, ...values);
     } else {
         return new LazyTranslatedString(term);
```
Comment (sdegueldre) [inline addons/web/static/src/core/l10n/translation.js:30]: You're not forwarding the values correctly here

Comment (oomsveta) [reply addons/web/static/src/core/l10n/translation.js]: whoops, fixed 🤠

**Review by xmo-odoo — APPROVED** [review]
Provisionally, no issue so far on the security front.

File: addons/web/static/tests/core/l10n/translation_tests.js [hunk 2]
```diff
@@ -171,3 +171,21 @@ QUnit.test("tamil has the correct numbering system", async (assert) => {
...
+
+QUnit.test(
+    "_t fills the format specifiers in lazy translated terms with its extra arguments",
+    async (assert) => {
+        translatedTerms[translationLoaded] = false;
+        const translatedStr = _t("Due in %s days", 513);
+        translatedTerms[translationLoaded] = true;
+        assert.equal(translatedStr, "Due in 513 days");
+    }
+);
```
Comment (sdegueldre) [inline addons/web/static/tests/core/l10n/translation_tests.js:191]: Please add a translation term for these to check that interpolation works in conjunction with translation

Comment (oomsveta) [reply addons/web/static/tests/core/l10n/translation_tests.js]: changes done

**Review by sdegueldre — APPROVED** [review]

File: addons/website_forum/static/src/js/website_forum.js [hunk 3]
```diff
@@ -264,7 +264,7 @@ publicWidget.registry.websiteForum = publicWidget.Widget.extend({
         ev.preventDefault();
         if (session.is_website_user) {
             this._displayAccessDeniedNotification(
-                Markup`<p>${sprintf(_t('Oh no! Please <a href="%s">sign in</a> to vote'), "/web/login")}</p>`
+                Markup`<p>${_t('Oh no! Please <a href="%s">sign in</a> to vote', "/web/login")}</p>`
```
Comment (sdegueldre) [inline addons/website_forum/static/src/js/website_forum.js:267]: @flch-odoo FYI, this doesn't work: the term being translated will not be extracted within the interpolation sigils of template strings, it should be translated outside the template string. (same thing for the other string further down in this file). This probably should be fixed in stable

Comment (flch-odoo) [reply addons/website_forum/static/src/js/website_forum.js:267]: @sdegueldre , thanks for the notice. Will do!

**Pull request closed without merging**